src/process/executable_path.cr

```diff
@@ -28,13 +28,45 @@ class Process
     end
   end
 
+  private def self.is_executable_file?(path)
+    unless File.info?(path, follow_symlinks: true).try &.file?
+      return false
+    end
+    {% if flag?(:win32) %}
+      true
+    {% else %}
+      File.executable?(path)
+    {% end %}
+  end
+
   # Searches an executable, checking for an absolute path, a path relative to
   # *pwd* or absolute path, then eventually searching in directories declared
   # in *path*.
   def self.find_executable(name : Path | String, path : String? = ENV["PATH"]?, pwd : Path | String = Dir.current) : String?
-    name = Path.new(name)
+    find_executable_possibilities(Path.new(name), path, pwd) do |p|
+      if is_executable_file?(p)
+        return p.to_s
+      end
+    end
+    nil
+  end
+
+  private def self.find_executable_possibilities(name, path, pwd)
+    return if "#{name}".empty?
```

Comment:
The interpolation can be avoided.

```suggestion
    return if name.to_s.empty?
```

Reply:
All done!

Yes for some reason I was afraid to explicitly call `to_s`, but that's what interpolation does directly anyway, so...

Reply:
Proof: https://github.com/crystal-lang/crystal/blob/aed3d6f5afa5bfc33600545ad49eba2f631263ac/src/string.cr#L5123-L5125